src/seo/getAlternateLanguageLinks.test.ts

```diff
@@ -0,0 +1,150 @@
+import { NewsroomLanguageSettings } from '@prezly/sdk';
+import { LocaleCode, LocaleObject } from '../intl';
+import { getAlternateLanguageLinks } from './getAlternateLanguageLinks';
+
+function lang(code: LocaleCode, isDefault = false) {
+    return { code, is_default: isDefault } as NewsroomLanguageSettings;
+}
+
+function getTranslationUrl(locale: LocaleObject) {
+    return `translationUrl/${locale.toHyphenCode()}`;
+}
+
+function createHref(locale: LocaleObject, translationUrl: string) {
+    return `http://localhost:3000/${locale.toNeutralLanguageCode()}/${translationUrl}`;
+}
+
+describe('getAlternateLanguageLinks', () => {
+    it('should handle one en-GB non-default language', () => {
+        const links = getAlternateLanguageLinks([lang('en-GB')], getTranslationUrl, createHref);
+
+        expect(links).toMatchObject([
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'en' },
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'en-GB' },
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'x-default' },
+        ]);
+    });
+
+    it('should handle one en-GB default language', () => {
+        const links = getAlternateLanguageLinks(
+            [lang('en-GB', true)],
+            getTranslationUrl,
+            createHref,
+        );
+
+        expect(links).toMatchObject([
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'en' },
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'en-GB' },
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'x-default' },
+        ]);
+    });
+
+    it('should handle two full locales', () => {
+        const links = getAlternateLanguageLinks(
+            [lang('en-GB', true), lang('fr-FR')],
+            getTranslationUrl,
+            createHref,
+        );
+
+        expect(links).toMatchObject([
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'en' },
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'en-GB' },
+            { href: 'http://localhost:3000/fr/translationUrl/fr-FR', hrefLang: 'fr' },
+            { href: 'http://localhost:3000/fr/translationUrl/fr-FR', hrefLang: 'fr-FR' },
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'x-default' },
+        ]);
+    });
+
+    it('should use en-GB locale as x-default even if other locale is set as default', () => {
+        const links = getAlternateLanguageLinks(
+            [lang('en-GB'), lang('fr-FR', true)],
+            getTranslationUrl,
+            createHref,
+        );
+
+        expect(links).toMatchObject([
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'en' },
+            { href: 'http://localhost:3000/en/translationUrl/en-GB', hrefLang: 'en-GB' },
```

Comment:
Is this valid?

I've checked this article on hreflang best practices here: https://ahrefs.com/blog/hreflang-tags/

They list this typical problem No. 3 in their article:

> ## 3. Page referenced for more than one language in hreflang
> 
> This warning triggers when one or more URLs are referenced for more than one language in hreflang annotations. For example:
> 
> ```
> <link rel="alternate" hreflang="en" href="http://example.com/page.html" />
> <link rel="alternate" hreflang="de" href="http://example.com/page.html" />
> ```
> 
> ### Why it’s an issue
> 
> Each piece of content should only serve one language or language-location. Having two or more contradicting references will confuse search engines, and they may end up ignoring both hreflang attributes.
> 
> ### How to fix
> 
> Review the affected pages, then inspect the URLs that reference the page in their hreflang attributes for errors. Remove the incorrect hreflang attribute to leave only one correct attribute per language.

Reply:
There are two completely different locales in the example, I use the same language part.
I tried https://codesandbox.io/s/confident-curran-esr8gk?file=/b.html here https://app.hreflang.org/ and there were no errors, however, there is indeed an error when different languages point to the same page

Reply:
> however, there is indeed an error when different languages point to the same page

Do you think it makes sense to build-in protection logic into the function to make sure we do not generate two alternate links with the same `href` for different `hreflang` values?

Reply:
I am not sure how that could happen since we construct links from languages array. There can be a possibility if server will return duplicated languages, but it is already covered by using "Map". I added test for that case.

Reply:
> but it is already covered by using "Map". I added test for that case.

I don't think I understood this part.

In these test scenarios, I can see that the code can generate multiple links with the same `href`, but different `hrefLag`, and it's not covered by using the Map. And according to the article I've linked above it is an issue.

Reply:
There is no error if the same href i used for different regions (or without region) of the same language https://github.com/prezly/theme-kit-nextjs/pull/215#discussion_r1104149754 
As i understand the problem will be if the same page is used for different languages like here
```
<link rel="alternate" hreflang="en" href="http://example.com/page.html" />
<link rel="alternate" hreflang="de" href="http://example.com/page.html" />
```
But I do not quite understand how can we get into this situation